### ISSUE TITLE
"Find All References" does not work on closed files in C language

### ISSUE THREAD

**Issue by smcdef** [microsoft/vscode-cpptools#9444]
Bug type: Language Service

**Describe the bug**
- OS and Version: macOS Monterey Version 12.4
- VS Code Version: 1.68.0 (Universal)
- C/C++ Extension Version: v1.10.5
- Other extensions you installed (and if the issue persists after disabling them): bug exists when disabling all other extensions.
- If using SSH remote, specify OS of remote machine: Linux
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. Use "Find all references" on a declaration while the file with the reference is closed.
<img width="1194" alt="image" src="https://user-images.githubusercontent.com/26999519/173554085-583e3b28-9581-486b-b970-28c869dffddb.png">

2. Open the file (mmio.c) with the reference. Use "Find all references".
<img width="1232" alt="image" src="https://user-images.githubusercontent.com/26999519/173554349-ab2ca483-2220-47a4-8a81-b8b3e20a65c9.png">

3. When I open a new file pic.c, then using "Find all references" also can find a new reference.
<img width="1249" alt="image" src="https://user-images.githubusercontent.com/26999519/173555376-e7ae6d3a-fccd-4da0-9fc3-9373f5819a82.png">

Seems that "Find all references" only works on opened files not workspace. This issue exists as well when searching a symbol across files with ⌘T or when "Go to Definition", I cannot search for a symbol or go to definition which is in a non-opened file. 

**Expected behavior**

**Code sample and logs**

- Code sample
Actually this issue exists on my any workspaces. Take the following code as an example.
```bash
git clone https://kernel.googlesource.com/pub/scm/linux/kernel/git/will/kvmtool.git
```
- Configurations in `c_cpp_properties.json`
```json
{
    "configurations": [
        {
            "name": "kvmtool",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [
            ],
            "configurationProvider": "ms-vscode.makefile-tools"
        }
    ],
    "version": 4
}
```

**Comment by sean-mcmanus** [comment]
@nostawjd Are you seeing other symptoms similar to https://github.com/microsoft/vscode-cpptools/issues/9431 ? If your issue doesn't repro with 1.9.8, then it would be a new regression on related to this older issue. The Find All References code hasn't changed recently, but there have been other changes to lower level systems like process creation which could impact this.

**Comment by sean-mcmanus** [comment]
Oops, ignore my previous comments about an "older issue" -- I didn't realize this was a new issue.

**Comment by sean-mcmanus** [comment]
@smcdef You have "configurationProvider": "ms-vscode.makefile-tools" so we get browse information from that extension -- if you run the C/C++: Log Diagnostics command after opening some file, look for a section labeled "Custom browse configuration" -- that should contain the folders for which workspace symbols will be parsed. I'm guessing the Makefile Tools extension may not be sending us the correct folders and/or it may need to be configured. You could try removing that configuration provider to see if fixes the issue (although you may need to change other settings to get configured correctly for IntelliSense). Also, if you can check if this repros with 1.9.8 of our extension that would help us determine if this is a regression with our extension or not.

**Comment by nostawjd** [comment]
@sean-mcmanus 
I'm not sure if the problem I see has anything to do with the include paths issue.... When I highlight a symbol in a file I've opened and use Find References from the right-click menu it never returns anything in the References pane.... At the top of the pane the bar just shows a blue animation sliding from right to left.... I can stop it searching with the button on the top right of the References pane.

I don't see anything reported in the Problems pane / window and I don't see any popups indicating errors. I have had a background process popup appear in the bottom right of the VSCode window indicating a search is in progress which I can dismiss, but doesn't give me any info.

From what I see VSCode updated on Friday 6/10  to this version and this problem appears to have started after that occurred. I was using Find References last week with no issues. I haven't added any plugins or changed my projects/files/VSCode settings. 

If there is someplace I can look to provide more details I'm happy to do so.... Just not sure what to add that would be helpful/informative.

**Comment by sean-mcmanus** [comment]
@nostawjd Is your issue fixed if you downgrade to 1.9.8 of the extension? There's an option to downgrade an extension version in the VS extension UI window.

**Comment by nostawjd** [comment]
@sean-mcmanus  I'll try that tomorrow and let you know if it works.

**Comment by smcdef** [comment]
> @smcdef You have "configurationProvider": "ms-vscode.makefile-tools" so we get browse information from that extension -- if you run the C/C++: Log Diagnostics command after opening some file, look for a section labeled "Custom browse configuration" -- that should contain the folders for which workspace symbols will be parsed. I'm guessing the Makefile Tools extension may not be sending us the correct folders and/or it may need to be configured. You could try removing that configuration provider to see if fixes the issue (although you may need to change other settings to get configured correctly for IntelliSense). Also, if you can check if this repros with 1.9.8 of our extension that would help us determine if this is a regression with our extension or not.
> 
> Also, we could potentially try the repro ourselves with the repro project you referenced.

Thanks very much. @sean-mcmanus It works when I removed "configurationProvider": "ms-vscode.makefile-tools". I don't know how this config is configured. Anyway, you have solve my problem. Thanks again.

**Comment by nostawjd** [comment]
@sean-mcmanus 
I reverted the C/C++ plugin to v1.9.8 and that fixes the issue I ran into with the Find References. Thank you for the suggestion!

**Comment by sean-mcmanus** [comment]
@nostawjd There is a known bug in which process creation can fail on Windows (https://github.com/microsoft/vscode-cpptools/issues/9431) so you may be hitting that, which should be fixed later today with 1.10.7. If your issue is different we'd probably need more info to fix it, since we're not aware of another issue yet.

**Comment by nostawjd** [comment]
@sean-mcmanus 
I'll look for the update tomorrow and try it out.... Will let you know if it works or not.

**Comment by sean-mcmanus** [comment]
@nostawjd We shipped 1.10.7 with the create process fix. Let us know if that solves the issue.

**Comment by nostawjd** [comment]
@sean-mcmanus I verified that 1.10.7 fixes the problem for me... Find References works.

Thank you!
